### PR TITLE
Migrates as many as possible test cases from Django mail module

### LINKED ISSUE #25: Add tests from django mail
Looking at https://github.com/django/django/tree/main/tests/mail they seem to have very decent test coverage. It would probably be a good idea to add those tests in here as well.

### PR DESCRIPTION
**Pull request by tkzt**
Totally, improved the test coverage from 63% to 94%.

fixes #25

### CODE REVIEW

File: tests/attachments/file.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+django/django
```
Comment (greyli) [inline tests/attachments/file.txt:1]: Maybe it's better to change it to `flask/flask`? :P

**Comment by greyli** [comment]
Could you move [this commit](https://github.com/waynerv/flask-mailman/pull/64/commits/1f902befed7bf1027840df3515513676a761423f) to a separate PR? Let's review and merge that first.

**Comment by tkzt** [comment]
> Could you move [this commit](https://github.com/waynerv/flask-mailman/pull/64/commits/1f902befed7bf1027840df3515513676a761423f) to a separate PR? Let's review and merge that first.

[Done~](https://github.com/waynerv/flask-mailman/pull/66)

**Comment by greyli** [comment]
Please resolve the conflicts and fix the lint issues:

```
  tests/test_backend.py:10:1: F401 'ssl.SSLError' imported but unused
  tests/test_mail.py:5:1: F401 'unittest.mock' imported but unused
  tests/test_mail.py:7:1: F401 'flask_mailman.utils.DNS_NAME' imported but unused
```

**Comment by tkzt** [comment]
> Please resolve the conflicts and fix the lint issues:
> 
> ```
>   tests/test_backend.py:10:1: F401 'ssl.SSLError' imported but unused
>   tests/test_mail.py:5:1: F401 'unittest.mock' imported but unused
>   tests/test_mail.py:7:1: F401 'flask_mailman.utils.DNS_NAME' imported but unused
> ```

Finished that 🍻

**Comment by greyli** [comment]
Merged, thanks!

**Pull request merged**